In [1]:
import os
import torch
from tensorboard.backend.event_processing import event_accumulator

HISTORY_PATHS = [f"history_v{i}.pt" for i in range(1, 16)]
EVENTS_DIRS = [f"events_v{i}" for i in range(1, 16)]


# ============
# history.pt
# ============
def read_history(history_path):
    print("\n=== HISTORY.PT ===")
    
    if not os.path.exists(history_path):
        print(f"[ERROR] {history_path}")
        return None

    history = torch.load(history_path, map_location="cpu", weights_only=False)

    for key, value in history.items():
        if isinstance(value, list):
            print(f"{key}: {value[-1] if value else 'no data'}")
        else:
            print(f"{key}: {value}")

    return history


# ============
# events.out.tfevents.*
# ============
def read_events(events_dir):
    print("\n=== TENSORBOARD EVENTS ===")

    events_files = [f for f in os.listdir(events_dir) if "tfevents" in f]

    if not events_files:
        print("[ERROR] 'events.out.tfevents.*'")
        return
    
    for fname in events_files:
        print(f"\n>>> File: {fname}")
        ea = event_accumulator.EventAccumulator(os.path.join(events_dir, fname))
        ea.Reload()

        tags = ea.Tags()
        print("Scalars:", tags.get("scalars", []))

        if "scalars" in tags:
            for tag in tags["scalars"]:
                print(f"\n--- {tag} ---")
                values = ea.Scalars(tag)
                for entry in values:
                    print(f"step={entry.step}, value={entry.value}")


if __name__ == "__main__":
    i = 0
    for history_path, events_dir in zip(HISTORY_PATHS, EVENTS_DIRS):
        i += 1
        print(f"\n############# VERSION {i} #############")
        read_history(history_path)
        #read_events(events_dir)


############# VERSION 1 #############

=== HISTORY.PT ===
train_loss: 0.21844465830632115
train_acc: 0.9902583871639925
val_loss: 0.4961155683460815
val_acc: 0.8379373848987108
test_loss: 0.6697330336238063
test_acc: 0.7252906976744186
f1_macro: no data
f1_weighted: no data
val_f1_macro: 0.8065353489082303
val_f1_weighted: 0.8069526591860144
test_f1_macro: 0.7246572296746461
test_f1_weighted: 0.7247340136746185

############# VERSION 2 #############

=== HISTORY.PT ===
train_loss: 0.2151233336718834
train_acc: 0.9912481767034799
val_loss: 0.7940505816297636
val_acc: 0.6703499079189686
test_loss: 0.8095367446888325
test_acc: 0.6933139534883721
f1_macro: no data
f1_weighted: no data
val_f1_macro: 0.8541776437674429
val_f1_weighted: 0.8548591560224902
test_f1_macro: 0.690197894503215
test_f1_weighted: 0.6903785356038039

############# VERSION 3 #############

=== HISTORY.PT ===
train_loss: 0.21597731547210086
train_acc: 0.9917170243800791
val_loss: 0.5897258581048217
val_acc: 0.784530386

+--------+------------+-----------+----------+---------+----------+---------+----------------+-----------------+----------------+-----------------+
| Version| train_loss | train_acc | val_loss | val_acc | test_loss| test_acc| val_f1_macro   | val_f1_weighted | test_f1_macro  | test_f1_weighted|
+--------+------------+-----------+----------+---------+----------+---------+----------------+-----------------+----------------+-----------------+
| 1      | 0.2184     | 0.9903    | 0.4961   | 0.8379  | 0.6697   | 0.7253  | 0.8065         | 0.8070          | 0.7247         | 0.7247          |
| 2      | 0.2151     | 0.9912    | 0.7941   | 0.6703  | 0.8095   | 0.6933  | 0.8542         | 0.8549          | 0.6902         | 0.6904          |
| 3      | 0.2160     | 0.9917    | 0.5897   | 0.7845  | 0.8126   | 0.6715  | 0.8225         | 0.8236          | 0.6638         | 0.6641          |
| 4      | 0.2127     | 0.9937    | 0.7506   | 0.7311  | 0.5931   | 0.7849  | 0.8576         | 0.8585          | 0.7846         | 0.7847          | <-- highest test_acc (N_FS)
| 5      | 0.2125     | 0.9935    | 0.6488   | 0.7735  | 0.6984   | 0.7253  | 0.8470         | 0.8475          | 0.7238         | 0.7239          |
| 6      | 0.2169     | 0.9907    | 0.5195   | 0.8048  | 0.6752   | 0.7355  | 0.8469         | 0.8475          | 0.7333         | 0.7334          |
| 7      | 0.2171     | 0.9917    | 0.5505   | 0.7864  | 0.7233   | 0.7195  | 0.8125         | 0.8141          | 0.7093         | 0.7096          |
| 8      | 0.2155     | 0.9922    | 0.5923   | 0.7716  | 0.8269   | 0.6686  | 0.8894         | 0.8897          | 0.6653         | 0.6655          | <-- lowest tes_acc (_CF_)
| 9      | 0.2136     | 0.9935    | 0.6856   | 0.7514  | 0.6234   | 0.7398  | 0.8301         | 0.8310          | 0.7350         | 0.7352          |
| 10     | 0.2123     | 0.9940    | 0.5782   | 0.7937  | 0.6218   | 0.7540  | 0.8602         | 0.8610          | 0.7421         | 0.7423          |
| 11     | 0.2119     | 0.9938    | 0.6127   | 0.7824  | 0.6305   | 0.7450  | 0.8515         | 0.8523          | 0.7312         | 0.7315          |
| 12     | 0.2141     | 0.9919    | 0.5950   | 0.7891  | 0.6452   | 0.7340  | 0.8450         | 0.8456          | 0.7285         | 0.7289          |
| 13     | 0.2130     | 0.9925    | 0.6034   | 0.7772  | 0.6375   | 0.7410  | 0.8489         | 0.8493          | 0.7321         | 0.7325          |
| 14     | 0.2126     | 0.9932    | 0.5898   | 0.7815  | 0.6221   | 0.7480  | 0.8520         | 0.8528          | 0.7360         | 0.7363          |
| 15     | 0.2118     | 0.9941    | 0.5734   | 0.7896  | 0.6087   | 0.7562  | 0.8583         | 0.8589          | 0.7425         | 0.7427          | <-- second highest test_acc (___S)
+--------+------------+-----------+----------+---------+----------+---------+----------------+-----------------+----------------+-----------------+